src/darwin/Framework/CHIP/templates/helper.js

```diff
@@ -36,6 +37,35 @@ function asObjectiveCBasicType(type, options)
   }
 }
 
+async function asTypedObjectiveCExpression(value, type)
```

Comment:
Please document, that this converts an expression involving possible variables whose types are objective C objects into an expression whose type is a C++ type?

And maybe this should be called `asTypedExpressionFromObjectiveC" or something, because the current name sounds like it's going to produce an Objective C value, which is not what's going on here.